fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexTestUtils.java

```diff
@@ -95,12 +253,322 @@ public class LuceneIndexTestUtils {
+                            field("complex").nest(function(LuceneFunctionNames.LUCENE_STORED, field("is_seen"))),
+                            field("complex").nest(function(LuceneFunctionNames.LUCENE_TEXT, field("text2"))),
+                            field("complex").nest(function(LuceneFunctionNames.LUCENE_STORED, field("score"))),
+                            field("complex").nest(function(LuceneFunctionNames.LUCENE_STORED, field("group"))),
+                            field("complex").nest(function(LuceneFunctionNames.LUCENE_SORTED, field("timestamp")))
+                    ), 8), LuceneIndexTypes.LUCENE,
+                    ImmutableMap.of(IndexOptions.TEXT_TOKENIZER_NAME_OPTION, AllSuffixesTextTokenizer.NAME,
+                            LuceneIndexOptions.OPTIMIZED_STORED_FIELDS_FORMAT_ENABLED, "true",
+                            LuceneIndexOptions.PRIMARY_KEY_SEGMENT_INDEX_ENABLED, "true"));
+
+    protected static final Index JOINED_COMPLEX_MULTIPLE_TEXT_INDEXES_NO_FREQS_POSITIONS =
+            new Index("JoinedComplex$text_multipleIndexes",
+                    concat(
+                            field("complex").nest(function(LuceneFunctionNames.LUCENE_STORED, field("is_seen"))),
+                            field("simple").nest(function(LuceneFunctionNames.LUCENE_TEXT, concat(field("text"),
+                                    function(LuceneFunctionNames.LUCENE_AUTO_COMPLETE_FIELD_INDEX_OPTIONS, value(LuceneFunctionNames.LuceneFieldIndexOptions.DOCS.name())),
+                                    function(LuceneFunctionNames.LUCENE_FULL_TEXT_FIELD_INDEX_OPTIONS, value(LuceneFunctionNames.LuceneFieldIndexOptions.DOCS.name()))))),
+                            field("complex").nest(function(LuceneFunctionNames.LUCENE_TEXT, field("text2"))),
+                            field("complex").nest(function(LuceneFunctionNames.LUCENE_STORED, field("score"))),
+                            field("complex").nest(function(LuceneFunctionNames.LUCENE_STORED, field("group"))),
+                            field("complex").nest(function(LuceneFunctionNames.LUCENE_SORTED, field("timestamp")))
+                    ), LuceneIndexTypes.LUCENE,
+                    ImmutableMap.of(IndexOptions.TEXT_TOKENIZER_NAME_OPTION, AllSuffixesTextTokenizer.NAME,
+                            LuceneIndexOptions.LUCENE_ANALYZER_NAME_OPTION, SynonymAnalyzer.QueryOnlySynonymAnalyzerFactory.ANALYZER_FACTORY_NAME,
+                            LuceneIndexOptions.TEXT_SYNONYM_SET_NAME_OPTION, EnglishSynonymMapConfig.ExpandedEnglishSynonymMapConfig.CONFIG_NAME,
+                            LuceneIndexOptions.OPTIMIZED_STORED_FIELDS_FORMAT_ENABLED, "true"));
+
+    public static Stream<Arguments> luceneIndexMapParams() {
+        return Stream.of(
+                Arguments.of(Map.ofEntries(
+                                Map.entry(SIMPLE_TEXT_SUFFIXES_KEY, SIMPLE_TEXT_SUFFIXES),
```

Comment:
The JOIN version of this seems much more complicated, and potentially more important, has the option `textSynonymSetName=EXPANDED_US_EN` which this does not.

Reply:
Actually going through these indexes, it looks like a lot of the join versions add a lot of fields, but I would expect them to just add the wrapping `field("complex").nest(...)` or `field("simple").nest(...)` around the expressions?